components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -149,33 +141,30 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
         this.attributes = attributes;
       });
 
-    // handle rule creation/update success
-    combineLatest([
-      this.store.select(createStatus),
-      this.store.select(updateStatus)
-    ]).pipe(
+    this.store.select(createStatus).pipe(
       takeUntil(this.isDestroyed),
-      filter(([cStatus, uStatus]) => this.saving && (!pending(cStatus) || !pending(uStatus))))
-      .subscribe(([cStatus, uStatus]) => {
-        this.saving = false;
-        if (cStatus === EntityStatus.loadingSuccess || uStatus === EntityStatus.loadingSuccess) {
-          this.closePage();
-        }
-      });
+      filter(status => this.saving && status === EntityStatus.loadingSuccess))
+      .subscribe(() => this.closePage());
+
+    this.store.select(updateStatus).pipe(
+      takeUntil(this.isDestroyed),
+      filter(status => this.saving && status === EntityStatus.loadingSuccess))
+      .subscribe(() => this.closePage());
 
-    // handle rule creation failure
     combineLatest([
       this.store.select(createStatus),
       this.store.select(createError)
     ]).pipe(
```

Comment:
The code here for add-project-rules is different than the other two scenarios, because we have one error that we want to show in the dialog.